modules/converting-machine-api-infra-machines.adoc

```diff
@@ -0,0 +1,46 @@
+// Module included in the following assemblies:
+//
+// * machine_management/creating-infrastructure-machines.adoc
+
+[id="converting-machine-api-infra-machines_{context}"]
+= Converting infrastructure machines created using the Machine API
+
+[IMPORTANT]
+====
+Creating infra machines using the Machine API is now deprecated.
+====
+
+Infrastructure machines created using the Machine API had both the `worker` and `infra` labels applied. This dual label required you to also taint the node so that compute workloads were not scheduled on it, to stay in compliance with our subscription guidelines. Managing infra workloads using taints is a tedious and cumbersome process, which is no longer recommended for infra nodes. Machines created using the Machine API must be converted to use only the `infra` label and have a custom machine config pool that manages it.
```

Comment:
It sounds like a product bug that you cannot simply create a machine set that has labels that the nodes apply to.  I would probably nack this change on this grounds alone.

Reply:
+1 For this being a product bug. From what I've been told, doc updates are being recommended to give a clear path forward until the issue can be fixed.

There was an alternative procedure proposed by @yuqi-zhang involving machine sets, although I haven't heard any other agreements on it:
>using machinesets + taint
a) create MCP
b) create machineset, with an added field that looks like:
>```
>taints:
>      - key: node-role.kubernetes.io/infra
>        effect: NoSchedule
>```
>Under `spec.template.spec.taints` of the machineset definition
c) once that node is up, strip worker label

I will hold on moving forward because the original doc plan no longer appears to be the right solution. @sferich888 @smarterclayton @yuqi-zhang @marrusl perhaps this should be discussed again by stakeholders to come up with a new plan for addressing this issue?

Reply:
@smarterclayton if we are treating this as a bug; can you work with @crawford and @yuqi-zhang to prioritize (I would like to push for blocker+ so we don't have a lingering issue) https://bugzilla.redhat.com/show_bug.cgi?id=1943579 (created a BZ for this issue) If so we can close https://issues.redhat.com/browse/RFE-1136 (and track this as a bug for the MCO team to correct).

Reply:
If we need something immediate to recommend to users in the current state of the product, we shouldn't block on my concerns.  I do expect the engineering teams to treat "inability to label machinesets OR machineconfig in a non-confilcting, clear and understandable way for humans" as a key feature that should have existed from 4.1 onwards (i.e. oops, bug).

Roughly, I expect the following:

1. MachineSets create nodes declaratively.
2. If I can't specify on a machine set that the nodes that the MachineSet creates have a specific label / taint, I can't declaratively create nodes.
3. MachineConfigs deal with nodes created from both machines and by humans
4. If I can't specify on a machine config pool that the machines in this pool have an additional set of labels, I can't declaratively tie on node behavior to labels
5. A node label created by a machineset takes priority over a node label created by a machineconfigpool and the machineconfigpool is responsible for not racing with the machineset

@crawford let's set up some time to work through the implications of this (I'm coming from the outside, I assume you have more context than I do).

Reply:
```
If I can't specify on a machine config pool that the machines in this pool have an additional set of labels, I can't declaratively tie on node behavior to labels
```

MCPs by design select based on labels, and does not attempt to apply labels to nodes. i.e. you add a node to a pool by labelling it (either via machineset or after the fact). The pool does not otherwise attempt to "create" nodes into it